next-update-cls/index.js

```diff
@@ -0,0 +1,359 @@
+            row["Shortlink"].replace("https://www.csa.gov.sg/", "") ===
+            itemPermalink
+        );
+        const category = csvRow["Product Category"];
+        const brand = csvRow["Brand"];
+        const model = csvRow["Model"];
+        const clsLevel = csvRow["CLS Level Issued"];
+        const registrationId = csvRow["CLS Label ID"];
+        const issuanceDate = moment(
+          csvRow["Date of Issue"],
+          "D-MMM-YY"
+        ).toDate();
+        const expirationDate = moment(
+          csvRow["Date of Expiry"],
+          "D-MMM-YY"
+        ).toDate();
+        const website = csvRow["Product URL"];
+        const vdp = csvRow["Vulnerability Disclosure Policy"];
+        const support = csvRow["Support Period"];
+        const product = csvRow["Product"];
+        const title = `${brand} ${model}`;
+
+        const productImage = csvRow["Product Image Filename"];
+        const labelImage = csvRow["Label Image Filename"];
+
+        // Upload the product and label images to S3
+        const productUuid = crypto.randomUUID();
+        const labelUuid = crypto.randomUUID();
+
+        await new Promise((resolve, reject) => {
+          exec(
+            `AWS_PROFILE=${AWS_PROFILE} aws s3 cp ${path.join(
+              "product",
+              productImage
+            )} s3://${S3_BUCKET_NAME}/${CSA_SITE_ID}/${productUuid}/${productImage}`,
+            (error, stdout, stderr) => {
+              if (error) {
+                console.error(`exec error: ${error}`);
+                reject(error);
+              }
+              resolve();
+            }
+          );
+        });
+
+        await new Promise((resolve, reject) => {
+          exec(
+            `AWS_PROFILE=${AWS_PROFILE} aws s3 cp ${path.join(
+              "label",
+              labelImage
+            )} s3://${S3_BUCKET_NAME}/${CSA_SITE_ID}/${labelUuid}/${labelImage}`,
+            (error, stdout, stderr) => {
+              if (error) {
+                console.error(`exec error: ${error}`);
+                reject(error);
+              }
+              resolve();
+            }
+          );
+        });
+
+        const productImageUrl = `/${CSA_SITE_ID}/${productUuid}/${productImage}`;
+        const labelImageUrl = `/${CSA_SITE_ID}/${labelUuid}/${labelImage}`;
+
+        // Step 4b: Create the article blob
+        const articleContent = getArticleSchema({
+          category,
+          brand,
+          model,
+          clsLevel,
+          registrationId,
+          issuanceDate,
+          expirationDate,
+          website,
+          vdp,
+          support,
+          productImage: productImageUrl,
+          labelImage: labelImageUrl,
+        });
+
+        // Create the new blob
+        const articleId = await client.query(
+          `INSERT INTO "Blob" (content) VALUES ($1) RETURNING id`,
+          [articleContent]
+        );
+
+        // Create the new resource
+        const newResource = await client.query(
```

Comment:
how are we deciding what to keep in tooling scripts vs monorepo? Would it make sense to consolidate such scripts under the monorepo to take advantage of the typesafety etc. Example, if tomorrow some columns or DB changes are done, we have to remember to change the scripts manually